Mathlib/GroupTheory/Submonoid/Basic.lean

```diff
@@ -93,6 +95,24 @@ structure Submonoid (M : Type*) [MulOneClass M] extends Subsemigroup M where
   one_mem' : (1 : M) ∈ carrier
 #align submonoid Submonoid
 
+/-- pull a submonoid back to an opposite submonoid along `unop`-/
+@[simps]
+def Submonoid.op {M : Type*} [MulOneClass M] (x : Submonoid M) : Submonoid (MulOpposite M) where
+  carrier := MulOpposite.unop ⁻¹' x.1
+  mul_mem' {a b} ha hb := by
+    simp only [Set.mem_preimage, MulOpposite.unop_mul, SetLike.mem_coe] at ha hb ⊢
+    exact mul_mem hb ha
+  one_mem' := Submonoid.one_mem' _
+
+/-- pull an opposite submonoid back to a submonoid along `op`-/
+@[simps]
+def Submonoid.unop {M : Type*} [MulOneClass M] (x : Submonoid (MulOpposite M)) : Submonoid M where
+  carrier := MulOpposite.op ⁻¹' x.1
+  mul_mem' {a b} ha hb := by
+    simp only [Set.mem_preimage, MulOpposite.unop_mul, SetLike.mem_coe] at ha hb ⊢
+    exact mul_mem hb ha
+  one_mem' := Submonoid.one_mem' _
```

Comment:
Can you put these in a new file to match https://github.com/leanprover-community/mathlib4/blob/3715499a32de6e3036c2bd3db4f94ff4bb0472c7/Mathlib/GroupTheory/Subgroup/MulOpposite.lean?

Reply:
done